### PR TITLE
Bugfix/set node

### PR DESCRIPTION
**Pull request by adisen**
**Description**
This PR allows Transforms.setNodes to take either a Partial<Node> as it second argument to props (which we have before) which overrides the props on the node or a function that takes the existing props for the relevant node and returns a new set of props to attach to the node.

**Issue**
Fixes: https://github.com/ianstormtaylor/slate/issues/3808

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)

### CODE REVIEW

File: packages/slate/src/transforms/node.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { update } from 'lodash'
```
Comment (timbuckley) [inline packages/slate/src/transforms/node.ts:1]: Ah it looks like we aren't using `update` yet in this code, maybe it's in code you haven't pushed yet?

Comment (adisen) [reply packages/slate/src/transforms/node.ts:1]: I didn't use it too 🤔

**Comment by adisen** [comment]
@timbuckley, can you please help me take a look at this? I'm not exactly sure what the problem is

**Comment by adisen** [comment]
> if you are talking about the failing test then the problem is in this file `docs/concepts/11-typescript.md`. Just to go this file and save it which will remove a line.
> That fixed the problem for me #4150

Thank you. But the file isn't part of this branch.

**Comment by princiya** [comment]
@adisen based on @nivekithan 's comment, you might want to refer to this [commit](https://github.com/ianstormtaylor/slate/pull/4150/commits/0430d4f21e73812c21fe8d496c7775aea5ef034a) where the `docs/concepts/11-typescript.md` had to be fixed despite not being used in the PR itself.

**Comment by adisen** [comment]
> @adisen based on @nivekithan 's comment, you might want to refer to this [commit](https://github.com/ianstormtaylor/slate/pull/4150/commits/0430d4f21e73812c21fe8d496c7775aea5ef034a) where the `docs/concepts/11-typescript.md` had to be fixed despite not being used in the PR itself.

I understand. But I can't even find the file in the repo, so there's no way of fixing it

**Comment by ianstormtaylor** [comment]
@adisen it might have gotten removed in subsequent changes. If you pull the latest changes to `main` and merge them into your branch (resolving any conflicts that occur) that should eliminate the error.

**Comment by adisen** [comment]
@timbuckley @BrentFarese @princiya  please take a look. Thanks

File: packages/slate/src/transforms/node.ts [hunk 2]
```diff
@@ -67,7 +68,7 @@ export interface NodeTransforms {
   ) => void
   setNodes: <T extends Node>(
     editor: Editor,
-    props: Partial<Node>,
+    props: Partial<T> | ((existingProps: NodeProps) => Partial<T>),
```
Comment (BrentFarese) [inline packages/slate/src/transforms/node.ts:71]: I would like to change the type of `NodeProps` to be generic.  Can you try the below and let me know if it works?

`type NodeProps<T extends Node> = T extends Element ? Omit<T, 'children'> : Omit<T, 'text'>;`

File: .github/ISSUE_TEMPLATE/bug-core.md [hunk 3]
```diff
@@ -1,13 +1,11 @@
 ---
+
```
Comment (BrentFarese) [inline .github/ISSUE_TEMPLATE/bug-core.md:2]: There is a lot of diff in this PR that you probably did not change?  Rebase off Main to get rid of the diff if you can.  Thanks.

File: .github/ISSUE_TEMPLATE/bug-core.md [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
 name: "\U0001F6A8 Bug: Core"
 about: A bug that occurs in Slate's core logic, on all platforms
 title: ''
 labels: bug
 assignees: ''
-
----
-
-**Description**
+---**Description**
```
Comment (princiya) [inline .github/ISSUE_TEMPLATE/bug-core.md:8]: we did a rebase together off the `main` branch. looks like the `yarn fix` command broke this? @adisen and I will check again...

File: packages/slate/src/transforms/node.ts (hunk 2 above)
Comment (adisen) [reply packages/slate/src/transforms/node.ts:71]: I'll change this in the interfaces/node.ts file right?

Comment (BrentFarese) [reply packages/slate/src/transforms/node.ts:71]: Wherever the type is defined change it there.  Thx.

**Comment by adisen** [comment]
I'm closing this cause of the rebasing issues. And I have opened another PR that solves the problem and keeps everything clean

**Pull request closed without merging**